File: python/DFS와 BFS(1260-s2).py
N, M, V = list(map(int, input().split()))

# 리스트 컴프리헨션 - 행렬 만들기
graph = [[0] * (N+1) for _ in range(N+1)]
for i in range(M):
    a, b = map(int, input().split())
    graph[a][b] = graph[b][a] = 1

dfs_visited_arr = [0] * (N+1)
bfs_visited_arr = [0] * (N+1)

# dfs - 재귀함수 이용
def dfs(V):
    dfs_visited_arr[V] = 1
    print(V, end=' ')
    for i in range(1, N+1):
        if (graph[V][i] == 1 and dfs_visited_arr[i] == 0):
            dfs(i)

# bfs - queue 이용
def bfs(V):
    queue = [V]
    bfs_visited_arr[V] = 1
    while queue:
        V = queue.pop(0)
        print(V, end=' ')
        for i in range(1, N+1):
            if (graph[V][i] == 1 and bfs_visited_arr[i] == 0):
                queue.append(i)
                bfs_visited_arr[i] = 1

dfs(V)
print()
bfs(V)

File: js/DFS와 BFS(1260-s2).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M, V] = input[0].split(' ').map(Number);
const graph = Array.from({length: N+1}, () => Array(N+1).fill(0));

for (let i=1; i <= M; i++) {
    const [a, b] = input[i].split(' ').map(Number);
    graph[a][b] = 1;
    graph[b][a] = 1;
}

const dfs_visited_arr = new Array(N+1).fill(0);
const bfs_visited_arr = new Array(N+1).fill(0);

function dfs(x) {
    dfs_visited_arr[x] = 1;
    process.stdout.write(x + ' ');
    for (let i=1; i <= N; i++) {
        if (graph[x][i] === 1 && dfs_visited_arr[i] === 0) {
            dfs(i);
        }
    }
}

function bfs(x) {
    const queue = [x];
    bfs_visited_arr[x] = 1;

    while (queue.length) {
        const y = queue.shift();
        process.stdout.write(y + ' ');

        for (let i=1; i <= N; i++) {
            if (graph[y][i] ===1 && bfs_visited_arr[i] === 0) {
                bfs_visited_arr[i] = 1;
                queue.push(i);
            }
        }
    }
}

dfs(V);
console.log();
bfs(V);
console.log();